src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -449,6 +446,35 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		}
 	}
 
+	private loadContentMetadata(metadata: INotebookMetadataInternal): void {
+		this._savedKernelInfo = metadata.kernelspec;
+		this._defaultLanguageInfo = metadata.language_info;
+		// If language info was serialized in the notebook, attempt to use that to decrease time
+		// required until colorization occurs
+		if (this._defaultLanguageInfo) {
+			this.updateLanguageInfo(this._defaultLanguageInfo);
+		}
+		this._savedConnectionName = metadata.connection_name;
+		this._multiConnectionMode = !!metadata.multi_connection_mode;
+
+		//Telemetry of loading notebook
+		if (metadata.azdata_notebook_guid && metadata.azdata_notebook_guid.length === 36) {
+			//Verify if it is actual GUID and then send it to the telemetry
+			let regex = new RegExp('(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}');
```

Comment:
There's a vscode utility for this. Can we use that instead? `isUUID(value: string)`